osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/Event.java

```diff
@@ -10,32 +10,37 @@
 
 import java.io.Serializable;
 
+import lombok.Getter;
 import org.joda.time.DateTime;
 
+@Getter
 public class Event implements Serializable {
 
-    /**
-     * Serial Version UID.
-     */
     private static final long serialVersionUID = 4482313912422705642L;
     private final DateTime timestamp;
     private final Integer eventCode;
-    private final Integer eventCounter;
     private final EventLogCategory eventLogCategory;
+    private final Integer eventCounter;
+    private final DateTime startTime;
+    private final Long duration;
 
     public Event(final DateTime timestamp, final Integer eventCode, final Integer eventCounter,
-            final EventLogCategory eventLogCategory) {
+            final EventLogCategory eventLogCategory, final DateTime startTime, final Long duration) {
         this.timestamp = timestamp;
         this.eventCode = eventCode;
         this.eventCounter = eventCounter;
         this.eventLogCategory = eventLogCategory;
+        this.startTime = startTime;
+        this.duration = duration;
     }
 
     @Override
     public String toString() {
-        return String.format("Event[time=%s, code=%s, category=%s, type=%s%s]", this.timestamp, this.eventCode,
+        return String.format("Event[time=%s, code=%s, category=%s, type=%s%s%s%s]", this.timestamp, this.eventCode,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1067)